packages/private-build-infra/src/addon-build-config-for-data-package.js

```diff
@@ -45,10 +46,34 @@ function addonBuildConfigForDataPackage(PackageName) {
       }
     },
 
-    _suppressCircularDependencyWarnings(message, next) {
-      if (message.code !== 'CIRCULAR_DEPENDENCY') {
-        next(message);
+    _suppressUneededRollupWarnings(message, next) {
+      if (message.code === 'CIRCULAR_DEPENDENCY') {
+        return;
+      } else if (message.code === 'NON_EXISTENT_EXPORT') {
+        // ignore ts-interface imports
+        if (message.message.indexOf(`/ts-interfaces/`) !== -1) {
+          return;
+        }
+      } else if (message.code === 'UNRESOLVED_IMPORT') {
+        if (!this.isDevelopingAddon()) {
+          // don't print these for consumers
+          return;
+        } else {
+          // make warning actionable
+          // eslint-disable-next-line no-console
+          console.log(
+            chalk.yellow(
+              `\n\n⚠️  Add ${chalk.white(
+                message.source
+              )} to the array returned by externalDependenciesForPrivateModule in index.js of ${chalk.white(
+                this.name
+              )}\n\n`
+            )
+          );
+          return;
```

Comment:
Instead of suppressing these, we should fail the build when developing. This will still have the same result (consumers will not see warnings), but actually forces us to ensure the repo is in the correct shape.

Reply:
@rwjblue I initially did make it throw but figured outputting a clear warning with specific guidance on how to fix it might be enough. I can still throw after this warning.

Reply:
Opened #6850 to address my concern.